projects/ngrx.io/content/guide/signals/signal-store/custom-store-features.md

```diff
@@ -1,3 +1,240 @@
 # Custom Store Features
 
-**UNDER CONSTRUCTION**
+Custom SignalStore features provide a robust mechanism for extending core functionality and encapsulating common patterns, facilitating reuse across multiple stores.
+
+## Creating a Custom Feature
+
+A custom feature is created using the `signalStoreFeature` function, which accepts a sequence of base or other custom features as input arguments and merges them into a single feature.
+
+### Example 1: Tracking Request Status
+
+The following example demonstrates how to create a custom feature that includes a `requestStatus` state property along with computed properties for checking the request status.
+
+<code-example header="request-status.feature.ts">
+
+import { computed } from '@angular/core';
+import { signalStoreFeature, withComputed, withState } from '@ngrx/signals';
+
+export type RequestStatus = 'idle' | 'pending' | 'fulfilled' | { error: string };
+export type RequestStatusState = { requestStatus: RequestStatus };
+
+export function withRequestStatus() {
+  return signalStoreFeature(
+    withState&lt;RequestStatusState&gt;({ requestStatus: 'idle' }),
+    withComputed(({ requestStatus }) => ({
+      isPending: computed(() => requestStatus() === 'pending'),
+      isFulfilled: computed(() => requestStatus() === 'fulfilled'),
+      error: computed(() => {
+        const status = requestStatus();
+        return typeof status === 'object' ? status.error : null;
+      }),
+    }))
+  );
+}
+
+</code-example>
+
+In addition to the state and computed properties, this feature also specifies a set of state updaters for modifying the request status.
+
+<code-example header="request-status.feature.ts">
+
+export function setPending(): RequestStatusState {
+  return { requestStatus: 'pending' };
+}
+
+export function setFulfilled(): RequestStatusState {
+  return { requestStatus: 'fulfilled' };
+}
+
+export function setError(error: string): RequestStatusState {
+  return { requestStatus: { error } };
+}
+
+</code-example>
+
+<div class="alert is-important">
+
+For a custom feature, it is recommended to define state updaters as standalone functions rather than feature methods. This approach enables tree-shaking, simplifies testing, and facilitates their use alongside other updaters in a single `patchState` call.
+
+</div>
+
+The `withRequestStatus` feature and updaters can be used to add the `requestStatus` state property, along with the `isPending`, `isFulfilled`, and `error` computed properties to the `BooksStore`, as follows:
+
+<code-example header="books.store.ts">
+
+import { inject } from '@angular/core';
+import { patchState, signalStore, withMethods } from '@ngrx/signals';
+import { setAllEntities, withEntities } from '@ngrx/signals/entities';
+import { setFulfilled, setPending, withRequestStatus } from './request-status.feature';
+import { Book } from './book.model';
+import { BooksService } from './books.service';
+
+export const BooksStore = signalStore(
```

Comment:
Should we show the interface of `BookStore` to display the merged result?
I think this could be useful to show how features impact the store.

Reply:
Good point Tim! I'll improve these examples in upcoming PR(s).